scripts/subset_hail_table.py

```diff
@@ -0,0 +1,169 @@
+    return ht
+
+
+def main(
+    ht_path: str,
+    output_root: str,
+    locus: hl.IntervalExpression | None,
+    biallelic: bool = False
+):
+    """
+
+    Parameters
+    ----------
+    ht_path : path to input MatrixTable
+    output_root : prefix for file naming
+    locus : an optional parsed interval for locus-based selection
+    biallelic : if True, filter the output MT to biallelic sites only
+
+    Returns
+    -------
+
+    """
+
+    ht = hl.read_table(ht_path)
+
+    if isinstance(locus, hl.IntervalExpression):
+        ht = subset_to_locus(ht, locus=locus)
+
+    if biallelic:
+        ht = ht.filter_rows(hl.len(ht.alleles) == 2)
+
+    # create the output path; make sure we're only ever writing to test
+    actual_output_path = output_path(output_root).replace(
+        f'cpg-{get_config()["workflow"]["dataset"]}-main',
+        f'cpg-{get_config()["workflow"]["dataset"]}-test',
+    )
+
+    # write the Table to a new output path
+    ht.write(f'{actual_output_path}.ht', overwrite=True)
+
+
+def clean_locus(contig: str, pos: str) -> hl.IntervalExpression | None:
+    """
+
+    Parameters
+    ----------
+    contig : the contig string, e.g. 'chr4'
+    pos : either a single coordinate or two in the form start-stop
+
+    Returns
+    -------
+    A parsed hail locus. For a point change this will be contig:pos-pos+1
+    """
+    if not any([contig, pos]):
+        return None
+
+    if pos and not contig:
+        raise Exception(f'Positional filtering requires a chromosome')
+
+    if contig and not pos:
+        start = 'start'
+        end = 'end'
+
+    elif '-' in pos:
+        assert (
+            pos.count('-') == 1
+        ), f'Positions must be one value, or a range between two values: {pos}'
+        start, end = pos.split('-')
+        if start != 'start':
+            assert int(start), f'start value could not be converted to an int: {start}'
+            if int(start) < 1:
+                start = 1
+        if end != 'end':
+            assert int(end), f'end value could not be converted to an int: {end}'
+            # adjust the end value if it is out of bounds
+            if int(end) > hl.get_reference('GRCh38').lengths[contig]:
+                end = hl.get_reference('GRCh38').lengths[contig]
+
+        # final check that numeric coordinates are ordered
+        if start != 'start' and end != 'end':
+            assert int(start) < int(end)
+
+    else:
+        assert int(
+            pos
+        ), f'if only one position is specified, it must be numerical: {pos}'
+        start = int(pos)
+        end = start + 1
+
+    return hl.parse_locus_interval(f'{contig}:{start}-{end}', reference_genome='GRCh38')
+
+
+if __name__ == '__main__':
+
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(asctime)s %(levelname)s %(module)s:%(lineno)d - %(message)s',
+        datefmt='%Y-%M-%d %H:%M:%S',
+        stream=sys.stderr,
+    )
+
+    parser = ArgumentParser()
```

Comment:
We generally use click for CLI, I think it's a bit cleaner. Not a big deal for this PR, but for new scripts :)